sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -194,12 +203,30 @@ class V2SessionCatalog(catalog: SessionCatalog)
   }
 
   override def dropTable(ident: Identifier): Boolean = {
+    dropTableInternal(ident)
+  }
+
+  private def dropTableInternal(ident: Identifier, purge: Boolean = false): Boolean = {
     try {
-      if (loadTable(ident) != null) {
+      val table = loadTable(ident)
+      if (table != null) {
```

Comment:
Hmm, is it possibly `loadTable` returning null? Isn't it always returning `V1Table`?

Reply:
Just follow old code 😂. Changed.

Reply:
> Isn't it always returning V1Table?

Please check https://github.com/apache/spark/pull/41348#discussion_r1221284308